wcfsetup/install/files/lib/system/user/authentication/oauth/User.class.php

```diff
@@ -0,0 +1,80 @@
+<?php
+namespace wcf\system\user\authentication\oauth;
+
+/**
+ * 
+ * 
+ * @author	Tim Duesterhus
+ * @copyright	2001-2021 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core\System\User\Authentication\Oauth
+ * @since	5.4
+ */
+final class User implements \ArrayAccess {
+	private $data = [];
+	
+	public function __construct($data) {
```

Comment:
Consider using paramters for `__id` and `__username`.

Reply:
And: Isn't the $data an array?

Reply:
I considered, but I am not using separate parameters here, because without named parameters it gets a bit wonky with the optional `__email` field (which needs to come last).